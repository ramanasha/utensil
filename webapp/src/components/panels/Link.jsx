import React, { Component } from 'react';

export default class Link extends Component {
    constructor(props) {
        super(props);
        this.handleClick = this.handleClick.bind(this);
    }

    handleClick() {
        window.open('http://' + this.props.url);
    }

    render() {
        return (
            <div className="link" onClick={this.handleClick}>
                {this.props.url.slice(4)}
            </div>
        );
    }
}
