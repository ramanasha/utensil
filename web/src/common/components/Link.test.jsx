import React from 'react';
import { mount } from 'enzyme';

import Link from './Link';

describe('<Link />', () => {
  it('should render properly', () => {
    const component = mount(<Link url='food.olin.edu' />);
    expect(component).toMatchSnapshot();
  });
});
