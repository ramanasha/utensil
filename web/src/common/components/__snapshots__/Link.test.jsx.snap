// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> should render properly 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <Link
    url="food.olin.edu"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "url": "food.olin.edu",
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="link"
      >
        food.olin.edu
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "food.olin.edu",
        "className": "link",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "food.olin.edu",
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "url": "food.olin.edu",
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="link"
        >
          food.olin.edu
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "food.olin.edu",
          "className": "link",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "food.olin.edu",
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
